include/LeastSquaresFit.hpp

```diff
@@ -4,22 +4,65 @@
 #ifndef LEASTSQUARESFIT_HPP
 #define LEASTSQUARESFIT_HPP
 #include "CSVParser.hpp"
-
+#include <functional>
 
 template <typename X_type, typename Y_type>
 class LeastSquaresFit;
 
-template <typename T>
+// a projection is the best guess with current data of the line of best fit
 struct projection{
-    T a_proj; // represents y-intercept
-    T b_proj; // represents slope
+    float_data_type a_proj; // represents y-intercept
+    float_data_type b_proj; // represents slope
 };
 
-template <typename T>
-projection<T> error( projection<T> const p1, projection<T> const p2);
+template<typename X_type, typename Y_type>
+struct point {
+	X_type x;
+	Y_type y;
+};
```

Comment:
I wonder if this is maybe a little _too_ general? Semantically, I have trouble imagining a "point" as having two different data types? Or, if this doesn't represent a "point", per se, maybe you could just use `std::tuple<>`?